dashboard/lib/projects_list.rb

```diff
@@ -13,13 +24,44 @@ def fetch_section_projects(section)
             # The channel id stored in the project's value field may not be reliable
             # when apps are remixed, so recompute the channel id.
             channel_id = storage_encrypt_channel_id(student_storage_id[:id], project[:id])
-            project_row_data = get_project_row_data(student, project, channel_id)
-            projects_list_data << project_row_data if project_row_data
+            project_data = get_project_row_data(student, project, channel_id)
+            projects_list_data << project_data if project_data
           end
         end
       end
     end
 
+    # Retrieve a hash of lists of published projects from the database, e.g.
+    #   {
+    #     applab: [{...}, {...}, {...}]
+    #   }
+    # when a single project type is requested, or the following when all types
+    # are requested:
+    #   {
+    #     applab: [{...}, {...}, {...}]
+    #     gamelab: [{...}, {...}, {...}]
+    #     playlab: [{...}, {...}, {...}]
+    #     artist: [{...}, {...}, {...}]
+    #   }
+    # @param project_type [String] Type of project to retrieve listed in
+    # PUBLISHED_PROJECT_TYPES, or 'all' to retrieve a list of each project type.
+    # @param limit [Integer] Maximum number of projects to retrieve of each type.
+    #   Must be between 1 and MAX_LIMIT, inclusive.
+    # @param offset [Integer] Number of projects to skip. Default: 0.
+    #   Must not specified when requesting all project types.
+    # @return [Hash<Array<Hash>>] A hash of lists of published projects.
+    def fetch_published_projects(project_type, limit, offset)
+      unless limit && limit.to_i >= 1 && limit.to_i <= MAX_LIMIT
+        raise ArgumentError, "limit must be between 1 and #{MAX_LIMIT}"
+      end
+      if project_type == 'all'
+        raise ArgumentError, 'Cannot specify offset when requesting all project types' if offset
+        return fetch_published_projects_all_types(limit)
+      end
+      raise ArgumentError, "invalid project type: #{project_type}" unless PUBLISHED_PROJECT_TYPES.include?(project_type)
+      [[project_type, fetch_published_projects_by_type(project_type, limit, offset)]].to_h
```

Comment:
I'm not sure which syntax you mean exactly, but I'm trying to do it as part of a map from array to object, which doesn't seem to be accounted for in that link. Is this any better?
```
      {}.tap do |projects|
        project_types.map do |type|
          projects[type] = fetch_published_project_type(type, limit)
        end
      end
```